solr-ingest/src/main/java/edu/unc/lib/dl/data/ingest/solr/filter/SetDescriptiveMetadataFilter.java

```diff
@@ -80,15 +85,36 @@ public void filter(DocumentIndexingPackage dip) throws IndexingException {
         }
 
         if (dip.getDocument().getTitle() == null) {
-            idb.setTitle(dip.getLabel());
+            idb.setTitle(getAlternativeTitle(dip));
         }
-        idb.getKeyword().add(dip.getPid().getPid());
+        idb.getKeyword().add(dip.getPid().getId());
+    }
+
+    private String getAlternativeTitle(DocumentIndexingPackage dip) throws FedoraException, IndexingException {
+        String title = null;
+        Resource resc = dip.getContentObject().getResource();
+
+        // Use dc:title as a default
+        if (resc.hasProperty(DcElements.title)) {
+            Statement dcTitle = resc.getProperty(DcElements.title);
+            return dcTitle.getString();
+        }
+
+        // fall back to filename if one is present
+        if (title == null && resc.hasProperty(Ebucore.filename)) {
+            // ebucore: filename
```

Comment:
Oops, I had pasted that there from the fedora ui as a reminder and forgot to remove it.  Good catch!